### PR TITLE
Core Profiler Passwordless Login

### PR DESCRIPTION
**Pull request by moon0326**
Related to #93160  #92105 

## Proposed Changes

* Applied new design from Figma
* Enabled passwordless login  (Password login still works)
* Enabled React version of `Lost your password?` page.

## Why are these changes being made?

* Apply the new design and enable passwordless for the core profiler customers.

## Known Issues

- The login email is not updated to reflect the new design. It's still using Jetpack template.
- After clicking the login link from the email, Jetpack login loading page is shwon.

### New login design

1. Setup [woocommerce-start-dev-env](https://github.com/woocommerce/woocommerce-start-dev-env) with this branch.
2. Create a new JN site with the latest WooCommerce. If you're redirected to the core profiler automatically, do not complete the core profiler. Go back to the admin area and copy the URL, admin username and password.
3. Open an incognito window and navigate to the JN site's admin area.
4. Click on `WooCommerce` from the menu to trigger the core profiler.
5. Proceed to the extensions page and make sure Jetpack is installed or selected. Continue.
6. You should be redirected to Jetpack connect page.
7. Click on `Log in` link.
8. Confirm the new login design.

Screenshot 2024-08-28 at 2 24 17 PM

### Lost your password 

1. Continue from `New login design` steps.
2. Click on `Lost your password?` link.
3. You should see the React version of `Forgot your password?` page.

### Login with password only account

1. Prepare an account that was created prior to the passwordless sign up.
2. Enter the username and click on `Continue`
3. The login page should add a password field.
4. Enter the password and click on `Continue` again.
5. You should be signed-in

### Login with passwordless account

1. Prepare an account that was created with passwordless sign up, but without any two factor auth types enabled.
2. Enter the username and click on `Continue`
3. You should see `Check your email` page. This design applies to all passwordless account types (sms, two factor, and security key)

Screenshot 2024-08-28 at 2 28 16 PM

4. Go to your email and click on the login link.
5. You should be signed-in.

### Login with passwordless + two factor account.

1. Prepare a passwordless + 2FA account.
2. Enter the username and click on `Continue`
3. You should see `Check your email` page.
4. GO to your email and click on the login link.
5. You should see 2FA verification page. 
Screenshot 2024-08-28 at 3 17 59 PM

6. Enter the verification code and click on `Continue`
7. You should be signed-in.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/login/magic-login/main-content-woo-core-profiler.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		startResendEmailCountdown();
+	};
+
+	useEffect( () => {
+		startResendEmailCountdown();
+
+		return () => {
+			resetResendEmailCountdown();
+		};
+	}, [] );
```
Comment (chihsuan) [inline client/login/magic-login/main-content-woo-core-profiler.tsx:62]: Should we include missing dependencies?

```suggestion
	}, [ dispatch, emailAddress, redirectToOriginal, token ] );
```

**Review by chihsuan — COMMENTED** [review]
Nice work on this! Tested the PR and functionality works as expected. I noticed a few things that need to be addressed regarding the design.

**Login page**

It may worth double-checking the font-size/color/font-weight of the elements in the design. Here are the differences I noticed:

1. Button font size is 14px in the design but it's 16px in the PR.
2. TOS text color is `#757575` in the design but it's `#646970` in the PR. Also font-family is `Inter` in the design. (Only TOS font-family is `Inter`, perhaps we should confirm with Veronica :thinking_face:)
3. `Lost your password?` text font size should be 13px.
4. Heading font family is `SF Pro` in the design but it's `proxima-nova, sans-serif` in the PR.
4. After logging to account, I can see the Connect your account page but it shows a different blue color when the page is loaded. (It only shows for a second and then it changes to the correct color)

Screenshot 2024-08-29 at 13 33 07

Screenshot 2024-08-29 at 13 49 17

Mobile view:

1. The `Don't have an account? Sign up` should be on the same line as the description text.
2. Padding between the masterbar and heading seems too much. It should be 24px.





**2FA**

1. The description text line-height/padding seems too much.
2. The button font-size is 14px in the design but it's 16px in the PR.

Mobile view:

1. The footer text should be left-aligned.




**You've got mail**

1. There is a ``Forgot password?` link in the design.
2. The button text should be `Log in`.

Screenshot 2024-08-29 at 13 52 48


**Check your email**

1. Heading font weight is 500 in the design but it's 700 in the PR.
2. `Open in...` button text font size is 14px in the design but it's 16px in the PR.


**Forgot password**

1. The hover color shows even when the button is disabled.


https://github.com/user-attachments/assets/03f32751-6549-427b-bbd7-156e2e9cf332

File: client/login/magic-login/main-content-woo-core-profiler.tsx (hunk 1 above)
Comment (moon0326) [reply client/login/magic-login/main-content-woo-core-profiler.tsx:62]: StartResendEmailCountdown should be called once at start. I think we don't have to pass any dependencies.

**Comment by moon0326** [comment]
@chihsuan Thank you for the detailed review! Appreciate it 🙇 

> Button font size is 14px in the design but it's 16px in the PR.
TOS text color is #757575 in the design but it's #646970 in the PR. Also font-family is Inter in the design. (Only TOS font-family is Inter, perhaps we should confirm with Veronica :thinking_face:)
Lost your password? text font size should be 13px.
Heading font family is SF Pro in the design but it's proxima-nova, sans-serif in the PR.

Fixed in [Style fixes for the "Log in to your account" desktop and mobile view](https://github.com/Automattic/wp-calypso/pull/93977/commits/d51e101f0aeb0c796a0bf03e53a0276d20905446)

> After logging to account, I can see the Connect your account page but it shows a different blue color when the page is loaded. (It only shows for a second and then it changes to the correct color)

This one is tricky. Both use `var(--wp-admin-theme-color)`. It looks like logged-out and logged-in have different theme color. I'll look into it.

> The Don't have an account? Sign up should be on the same line as the description text.
Padding between the masterbar and heading seems too much. It should be 24px.

Fixed in [Put sign up text at the same line](https://github.com/Automattic/wp-calypso/pull/93977/commits/7f908a7e3cbb96c07fe774bc34c00d1664caa833) 

> The description text line-height/padding seems too much.
The button font-size is 14px in the design but it's 16px in the PR.

Fixed in [Adjust text line-height on 2FA](https://github.com/Automattic/wp-calypso/pull/93977/commits/f49fce926a48b7cbb9f74a814c9b5a1be995ac25). Button size should be fixed in [Style fixes for the "Log in to your account" desktop and mobile view](https://github.com/Automattic/wp-calypso/pull/93977/commits/d51e101f0aeb0c796a0bf03e53a0276d20905446)

> You've got mail
There is a ``Forgot password?` link in the design.
The button text should be Log in.

Fixed in [Render forgot password link when password field is displayed](https://github.com/Automattic/wp-calypso/pull/93977/commits/9756aa9d215b9404482d0dabec2210f9b32cd97a) and [Change button text to "log in"](https://github.com/Automattic/wp-calypso/pull/93977/commits/d456194f101c39e629be5179e31118fe74dbc498)

> Check your email
Heading font weight is 500 in the design but it's 700 in the PR.
Open in... button text font size is 14px in the design but it's 16px in the PR.

Fixed in [Check your email page css fixes](https://github.com/Automattic/wp-calypso/pull/93977/commits/e69dd8dafc1470b12ab54624a4d0368a5c33c9ba)

> Forgot password
The hover color shows even when the button is disabled.

Fixed in [Fix disabled button hover state](https://github.com/Automattic/wp-calypso/pull/93977/commits/ce96dfcc7f3453846d663b6b26efd17ab3aa7a37)

File: client/login/magic-login/main-content-woo-core-profiler.tsx (hunk 1 above)
Comment (chihsuan) [reply client/login/magic-login/main-content-woo-core-profiler.tsx:62]: Gotcha! Then, how about adding a comment to it?

Comment (moon0326) [reply client/login/magic-login/main-content-woo-core-profiler.tsx:62]: @chihsuan Good idea! I'll add a document.

**Comment by moon0326** [comment]
Hey @chihsuan ✋ 

Could you take a look at the changes please 🙏 ?

File: client/blocks/login/index.jsx [hunk 2]
```diff
@@ -1001,6 +1003,10 @@ class Login extends Component {
 				sendMagicLoginLink={ this.sendMagicLoginLink }
 				isSendingEmail={ this.props.isSendingEmail }
 				isSocialFirst={ isSocialFirst }
+				loginButtons={ loginButtons }
```
Comment (chihsuan) [inline client/blocks/login/index.jsx]: Hey @moon0326 I got a white screen when I clicked on "Log in". It seems that `loginButtons` is undefined. Could you please check it? 

Screenshot 2024-09-03 at 11 21 07

Comment (moon0326) [reply client/blocks/login/index.jsx]: Hey @chihsuan  ✋ 

I wasn't able to reproduce it with the following steps. Could you provide reproducible steps please 🙏 ?

1. Navigated to https://wordpress.com/jetpack/connect/authorize?client_id=236381464&redirect_uri=https%3A%2F%2Fnoisily-fortunate-toucan.jurassic.ninja%2Fwp-admin%2Fadmin.php%3Fhandler%3Djetpack-connection-webhooks%26action%3Dauthorize%26_wpnonce%3D32ab4c9508%26redirect%3Dhttps%253A%252F%252Fnoisily-fortunate-toucan.jurassic.ninja%252Fwp-admin%252Fadmin.php%253Fpage%253Dwc-admin&state=1&scope=administrator%3A2ca8ea2582da1c066074563a12cc8b22&user_email=moon.kyong%40automattic.com&user_login=moon0326&jp_version=13.7&secret=QFoaAXaOIPZTr9p6Z5Hqf6RWkCboPylJ&blogname=Noisily+Fortunate+Toucan&site_url=https%3A%2F%2Fnoisily-fortunate-toucan.jurassic.ninja&home_url=https%3A%2F%2Fnoisily-fortunate-toucan.jurassic.ninja&site_icon&site_lang=en_US&site_created=2024-08-26+19%3A40%3A51&allow_site_connection=1&calypso_env=production&source&_as=search-term&_ak=jetpack&from=woocommerce-core-profiler&installed_ext_success=1&_ui=254662545&_ut=wpcom%3Auser_id&purchase_nonce=lbtPaOlO&_wp_nonce=69744335dc&redirect_after_auth=https%3A%2F%2Fnoisily-fortunate-toucan.jurassic.ninja%2Fwp-admin%2Fadmin.php%3Fpage%3Dwc-admin&site=https%3A%2F%2Fnoisily-fortunate-toucan.jurassic.ninja
2. Clicked `Log in`

Comment (chihsuan) [reply client/blocks/login/index.jsx]: Hey @moon0326,  I'm still getting a white screen when I click on "Log in." I used the same steps you provided. Here is the video recording. I can also see that `loginButtons` is undefined in my editor and confirm I have pulled the latest changes.  🤔 

https://github.com/user-attachments/assets/9095cb1e-51f4-4ebc-a7e4-695ebcd3d858

Comment (moon0326) [reply client/blocks/login/index.jsx]: Thanks @chihsuan 

I pulled the latest changes locally and I can reproduce it now. My apologies 🙏 I should have tested again after rebase.

Comment (moon0326) [reply client/blocks/login/index.jsx]: @chihsuan 

Fixed it in https://github.com/Automattic/wp-calypso/pull/93977/commits/816b62ec67e17a315b9483ffd4c938fb7ba0512e

We no longer pass `loginButtons` according to https://github.com/Automattic/wp-calypso/pull/93410/files#diff-a2085484a90c3affa18d524a40f82a2dcd6f62819f3c9246abdb4dca23feb5e2L1035 change.

**Review by chihsuan — COMMENTED** [review]
Thanks for addressing feedback! @moon0326 

Just found a few more things to address:

1. 2FA buttons seems too close to the edge of `OR` divider
2. When I clicked on "Sign up" in the 2FA screen, I was redirected to Jetpack Signup screen, not Core Profile Signup screen

Screenshot 2024-09-05 at 10 10 02


Not related to this PR, but I think we should create an issue to customize the "Email me a login link" screen?

Screenshot 2024-09-05 at 10 16 52


Other than that, everything looks good to me. Thanks for the hard work on this! 👍 

I'll pre-approve this PR.

**Review by chihsuan — APPROVED** [review]

**Comment by moon0326** [comment]
Thank you so much @chihsuan 🙇 

> 2FA buttons seems too close to the edge of OR divider

I think that's new 🤔 I'll double check other areas to make sure rebase did not cause any layout issues.

> When I clicked on "Sign up" in the 2FA screen, I was redirected to Jetpack Signup screen, not Core Profile Signup screen

Thank you for catching it 👍 

> Not related to this PR, but I think we should create an issue to customize the "Email me a login link" screen?

Yup! We should. I'll create a new issue

**Pull request merged**